src/java/org/apache/cassandra/db/Directories.java

```diff
@@ -799,7 +799,7 @@ public boolean accept(File file)
      * @return  Return a map of all snapshots to space being used
      * The pair for a snapshot has size on disk and true size.
      */
-    public Map<String, Pair<Long, Long>> getSnapshotDetails()
+    public synchronized Map<String, Pair<Long, Long>> getSnapshotDetails()
```

Comment:
How do you feel about creating a separate object for this, as we've done for compactions?

I.e., private static final Object SNAPSHOT_LOCK = new Object(); 

Then doing 

```
public Map<String, Pair<Long, Long>> getSnapshotDetails() {
   synchronized(SNAPSHOT_LOCK) { ... }
}
```


IIRC, the object that's used for locking for `synchronized` is itself, but if it's static (clearSnapshots), then it defaults to the class. I'd also imagine we want this to lock across threads?

Reply:
May also looking into if we can do a read-write style lock; calling listSnapshots from multiple threads is fine, it's only when we clear it becomes an issue.

Reply:
ah yeah you're right, i'll look into that